diff --git a/mk/Makefile b/mk/Makefile
new file mode 100644
index 0000000..a151bd9
--- /dev/null
+++ b/mk/Makefile
@@ -0,0 +1,49 @@
+REPONAME := ovmf
+include $(B_BASE)/common.mk
+include $(B_BASE)/rpmbuild.mk
+
+-include $(MY_OBJ_DIR)/version.inc
+$(MY_OBJ_DIR)/version.inc: $(REPOSTAMP)
+	rm -f $(MY_OBJ_DIR)/version.inc
+	$(call git_cset_number,$(REPONAME)) >> $@
+
+OVMF_VERSION := 0.1
+OVMF_REVISION := $(shell expr $(CSET_NUMBER))
+
+OVMF_SRPM := ovmf-$(OVMF_VERSION)-$(OVMF_REVISION).src.rpm
+SRPM := $(RPM_SRPMSDIR)/$(OVMF_SRPM)
+
+SOURCES := $(RPM_SOURCESDIR)/ovmf-$(OVMF_VERSION).tar.gz
+SOURCES += $(RPM_SPECSDIR)/ovmf.spec
+
+.PHONY: build
+build: $(MY_OBJ_DIR)/.rpmbuild.ovmf.stamp $(MY_SOURCES)/MANIFEST
+        @ :
+
+.PHONY: clean
+clean:
+	rm -f $(MY_OBJ_DIR)/.rpmbuild.ovmf.stamp
+	rm -f $(SOURCES)
+	rm -f $(MY_SOURCES)/MANIFEST
+
+$(RPM_SOURCESDIR)/ovmf-$(OVMF_VERSION).tar.gz: $(RPM_SOURCESDIR)/.dirstamp
+	{ set -e; set -o pipefail; \
+	cd $(call git_loc,$(REPONAME)); \
+	git archive --format=tar --prefix=$(REPONAME)-$(OVMF_VERSION)/ HEAD | gzip > $@.tmp; \
+	mv -f $@.tmp $@; \
+	}
+
+$(RPM_SPECSDIR)/ovmf.spec: ovmf.spec.in $(RPM_SPECSDIR)/.dirstamp
+	{ set -e; set -o pipefail; \
+	sed -e 's/@OVMF_VERSION@/$(OVMF_VERSION)/g' \
+	    -e 's/@OVMF_REVISION@/$(OVMF_REVISION)/g' \
+		< $< > $@.tmp; \
+	mv -f $@.tmp $@; \
+	}
+
+$(MY_OBJ_DIR)/.rpmbuild.ovmf.stamp: $(SOURCES) $(RPM_DIRECTORIES)
+	$(RPMBUILD) -ba $(RPM_SPECSDIR)/ovmf.spec
+	touch $@
+
+$(MY_SOURCES)/MANIFEST: $(MY_SOURCES)/.dirstamp
+	echo "$(COMPONENT) $(call rpm_license,$(SRPM)) file $(SRPM)" >$@
diff --git a/mk/ovmf.spec.in b/mk/ovmf.spec.in
new file mode 100644
index 0000000..ad096f3
--- /dev/null
+++ b/mk/ovmf.spec.in
@@ -0,0 +1,36 @@
+Name: ovmf
+Summary: OVMF XXX
+Version: @OVMF_VERSION@
+Release: @OVMF_REVISION@
+
+Group: Development/Tools
+License: GPLv2
+URL: http://xenbits.xen.org/gitweb/?p=ovmf.git
+Source0: %{name}-%{version}.tar.gz
+
+BuildRequires: nasm
+
+
+%description
+OVMF XXX
+
+
+%prep
+%setup -q
+
+
+%build
+OvmfPkg/build.sh -D SECURE_BOOT_ENABLE -a X64 -b DEBUG -n %{?_smp_flags}
+
+
+%install
+install -m 755 -d %{buildroot}/%{_datadir}/%{name}
+install -m 644 Build/OvmfX64/DEBUG_GCC*/FV/OVMF_CODE.fd %{buildroot}/%{_datadir}/%{name}/ovmf.bin
+install -m 644 Build/OvmfX64/DEBUG_GCC*/FV/OVMF_VARS.fd %{buildroot}/%{_datadir}/%{name}/ovmf-vars.bin
+
+
+%files
+%{_datadir}/%{name}
+
+
+%changelog
