diff --git a/OvmfPkg/OvmfPkgX64.fdf b/OvmfPkg/OvmfPkgX64.fdf
index db3624c..ead8b7a 100644
--- a/OvmfPkg/OvmfPkgX64.fdf
+++ b/OvmfPkg/OvmfPkgX64.fdf
@@ -306,6 +306,12 @@ INF MdeModulePkg/Logo/LogoDxe.inf
     SECTION PE32 = Intel3.5/EFIX64/E3522X2.EFI
   }
 !endif
+  FILE DRIVER = 9C1011A2-FC75-4C61-BFF1-AFAACA86DCD0 {
+    SECTION PE32 = 10ec8139.efi
+  }
+  FILE DRIVER = C4F14A61-D11E-499E-AE53-3CC64428E55A {
+    SECTION PE32 = 8086100e.efi
+  }
   INF  MdeModulePkg/Universal/Network/SnpDxe/SnpDxe.inf
   INF  MdeModulePkg/Universal/Network/DpcDxe/DpcDxe.inf
   INF  MdeModulePkg/Universal/Network/MnpDxe/MnpDxe.inf
